### ISSUE TITLE
Softmax core enabled on DPUCZDX8G, but vai_c_tensorflow doesn't recognize it

### ISSUE THREAD

**Issue by anirudhitagi** [Xilinx/Vitis-AI#904]
Greetings,

I am trying to deploy [tf_mobilenetv2_1.4_imagenet_224_224_1.16G_2.5](https://github.com/Xilinx/Vitis-AI/tree/master/model_zoo/model-list/tf_mobilenetv2_1.4_imagenet_224_224_1.16G_2.5) on an ultra96 v2  with Vitis-AI (latest version, Docker build: 2.5.0.1260).

I am following the Tensorflow process. I am able to successfully calibrate and quantize the model (with **vai_q_tensorflow**). 
I generated my own arch.json with Softmax core enabled. I am using the latest DPU v4.0.Here are the configurations of the DPU:
![image](https://user-images.githubusercontent.com/29096562/177307633-963dc652-ccf4-4c01-a146-1d8ad0ddcce7.png)
![image](https://user-images.githubusercontent.com/29096562/177307714-639d97f8-b2aa-49df-ae68-9676dcb83c53.png)
![image](https://user-images.githubusercontent.com/29096562/177307762-318ae59e-73f0-4abd-95c9-7999ed646985.png)
Here is the generated arch.json fingerprint for reference: **{"fingerprint":"0x101000037010404"}**

I compiled the model with vai_c_tensorflow, taking the above mentioned arch.json. While the compilation is successful (with no warning or messages), the .xmodel is still assigning the Softmax computation to the CPU. I noticed this by running :
 **xdputil xmodel {modelname}.xmodel -l**.
Please see the results below:

**{
    "subgraphs":[
        {
            "index":0,
            "name":"subgraph_input",
            "device":"USER"
        },
        {
            "index":1,
            "name":"subgraph_MobilenetV2/Conv/Conv2D",
            "device":"DPU",
            "fingerprint":"0x101000037010404",
            "DPU Arch":"DPUCZDX8G_ISA1_B1600_0101000037010404",
            "workload":1174662713,
            "input_tensor":[
                {
                    "index":0,
                    "name":"input/aquant",
                    "shape":[
                        1,
                        224,
                        224,
                        3
                    ],
                    "fixpos":6
                }
            ],
            "output_tensor":[
                {
                    "index":0,
                    "name":"MobilenetV2/Predictions/Reshape/aquant",
                    "shape":[
                        1,
                        1001
                    ],
                    "fixpos":4
                }
            ],
            "reg info":[
                {
                    "name":"REG_0",
                    "context type":"CONST",
                    "size":6086601
                },
                {
                    "name":"REG_1",
                    "context type":"WORKSPACE",
                    "size":1154048
                },
                {
                    "name":"REG_2",
                    "context type":"DATA_LOCAL_INPUT",
                    "size":150528
                },
                {
                    "name":"REG_3",
                    "context type":"DATA_LOCAL_OUTPUT",
                    "size":1008
                }
            ]
        },
        {
            "index":2,
            "name":"subgraph_MobilenetV2/Predictions/Reshape/aquant_MobilenetV2/Predictions/Softmax",
            "device":"CPU"
        }
    ]
}**

As per my understanding it is still inferring the Softmax computation to happen on the CPU (please refer "index":2 above ).

1. why is the vai_c_tensorflow not recognizing the softmax core and assigning softmax computation to the DPU?

2. This is the vai_c_tensorflow command I am giving:
**vai_c_tensorflow -f ./quantized/quantize_eval_model.pb -a {new_archjson_path}/arch.json -o ./compiled -n mobilenetv2_tf
**
do I have to add some arguments/options to this command? If so please advise which arguments/options.

3. Is there a way to extract the DPU configurations from the fingerprint or from the xmodel? It would help to doubly confirm that the DPU configurations are given correctly.

Please let me know if need to provide anymore information. Any help would be greatly appreciated! Thanks in advance.

Thanks and Regards,
-AI

**Comment by qianglin-xlnx** [comment]
HI @anirudhitagi 
It's expected. Compared with DPU, Softmax IP is another acceleration IP.
Softmax OP is not supported by DPU. So it's assigned to CPU when you use "vai_c_tensorflow" to compile the model.

**Comment by anirudhitagi** [comment]
Hi @qianglin-xlnx ,
Thank you for your reply! 
If softmax operation is not supported in the DPU, then what exactly is the purpose of SFM Cores?
What do I have to do if I want to use the softmax core inside the DPU?

**Comment by qianglin-xlnx** [comment]
Hi @anirudhitagi 
You can implement the Softmax IP as plugin, refer to:
https://github.com/Xilinx/Vitis-AI/tree/master/src/Vitis-AI-Runtime/VART/plugin-samples/samples/plugin-smfc
You can also call the Softmax IP as an runner, refer to: 
https://github.com/Xilinx/Vitis-AI/blob/master/src/Vitis-AI-Runtime/VART/vart/dpu-runner/samples/resnet50_zero_copy/resnet50.cpp#L189

**Comment by anirudhitagi** [comment]
Thank you for your response @qianglin-xlnx !
I will try ypur suggestion and update the results here.

Thanks and Regards,
-AI

**Comment by qianglin-xlnx** [comment]
Hi @anirudhitagi 
Is there any update?

**Comment by qianglin-xlnx** [comment]
Hi @anirudhitagi 
Since we haven't received your reply for a long time, we assume you have solved this issue and I'm going to close it. If you still have any questions, please feel free to reopen it. Thank you very much.
